cogs/moderation.py

```diff
@@ -293,6 +298,96 @@ async def clear(self, ctx: Context, amount: int) -> None:
         await asyncio.sleep(2.5)
         await message.delete()
 
+    @command()
+    @has_permissions(manage_messages=True)
+    async def shift(self, ctx, count: int, target: TextChannel, copy: bool = False) -> None:
+        """Copy or Move specified messages amount to specified channel"""
+        messages = []
+        async for message in ctx.message.channel.history(limit=count):
+            embed = Embed(description=message.content, color=Color.green())
+            embed.set_author(name=message.author.name, icon_url=message.author.avatar_url)
+            embed.timestamp = message.created_at
+            messages.append(embed)
+
+            if not copy:
+                await message.delete()
+
+        await target.send(f'Message source : {ctx.channel.mention}.')
+
+        for embed in reversed(messages):
+            await target.send(embed=embed)
+
+    @command()
+    @has_permissions(manage_messages=True)
+    async def dm(self, ctx: Context, members: Greedy[Member], *, message: str) -> None:
+        """Dm a List of Specified User from Your Guild."""
+        embed = Embed(
+            title="Notice!",
+            description=message,
+            color=Color.dark_magenta()
+        )
+        embed.set_footer(text=f"From {ctx.guild.name}", icon_url=ctx.guild.icon_url)
+
+        for member in members:
+            await member.send(embed=embed)
+
+    @command()
+    @has_permissions(administrator=True)
+    async def dmall(self, ctx: Context, *, message: str) -> None:
+        """Dm all Users from Your Guild."""
+        embed = Embed(
+            title="Notice!",
+            description=message,
+            color=Color.dark_magenta()
+        )
+        embed.set_footer(text=f"From {ctx.guild.name}", icon_url=ctx.guild.icon_url)
+
+        for member in ctx.guild.members:
+            with suppress(Forbidden):
+                await member.send(embed=embed)
+
+    @command()
+    @has_permissions(manage_roles=True)
```

Comment:
We shouldn't check for `manage_roles` for this, but instead, manage permissions for that given channel together with manage messages, that makes much more sense, why would a user need to be able to manage roles in order to lock a channel.

Note: This also applies for `unlock`.

Reply:
uhm, its not possible.